City-of-Helsinki/drupal-hdbt
# UHF-3884 Sidebar

# Sidebar [UHF-3884](https://helsinkisolutionoffice.atlassian.net/browse/UHF-3884)
Refactor the sidebar visibility logic on TPR Unit, TPR Service and standard pages

## What was done
* Added 'active_theme' variable for page preprocess to support sub theme blocks in page templates.
* Refactored sidebar visibility logic for TPR Unit, TPR Service and standard page.
* Added new variables for page.html.twig (in_menu, active_theme, has_sidebar).
   * Converted page_has_sidebar variables to use has_sidebar variable. 
   * Converted has_navigation variables to use in_menu variable. 
   * Refactored move_before_content logic
* Refactored TPR Unit and TPR Service templates based on refactored "in_menu" variable.
* Fixed margins for main-content as the basic page title block render logic has changed.

## How to install and test
* Get the latest db from Sote
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Log in as a non-uid1 user f.e. `drush uli --uid=10`
   * Go to these unpublished pages and check that the sidebar navigation is on either (-->) right side of the page or it's missing altogether 
      * TPR Service: https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/terveydenhoito/hammashoito/erikoishammaslaakarin-vastaanotto
      * TPR Unit: https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/terveydenhoito/hammashoito/hammashoitolat/jakomaen-hammashoitola
      * Standard page: https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/toihin-meille/opiskelijoille
      
* Update the necessary modules and themes
    * `composer require drupal/helfi_platform_config:dev-UHF-3884_sidebar && composer require drupal/hdbt_admin:dev-UHF-3884_sidebar && composer require drupal/hdbt:dev-UHF-3884_sidebar`
* Run `make drush-updb shell`
   * In shell run: `drush locale:check; drush locale:update; drush cr` 
   * Go to the unpublished pages listed above and check that the sidebar navigation is on (<--) left side of the page. 
   * As the "opiskelijoille" page is rendered without the sidebar navigation in [production](https://www.hel.fi/fi/sosiaali-ja-terveyspalvelut/toihin-meille/opiskelijoille), we need to be able to hide the sidebar navigation manually on chosen pages. For this to happen, go and edit the "opiskelijoille" page. 
      * There should be a new field "hide sidebar navigation from this page" inside the "Menu settings" (in node and tpr unit/service forms)
      * ![image](https://user-images.githubusercontent.com/1712902/163395451-e1025de4-7c60-4038-a94a-06b7d1cafe28.png)
      * Select the checkbox, save the node
      * Check that the sidebar navigation has disappeared from the page.

## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt-admin/pull/129
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/306
